src/main/java/org/mule/extension/api/introspection/property/StudioModelProperty.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.api.introspection.property;
+
+/**
+ * Represents an extension's capability to provide a custom UI description
+ *
+ * @since 1.0
+ */
+public interface StudioModelProperty
+{
+
+    /**
+     * A unique key to identify this model property from others
+     */
+    String KEY = "STUDIO.EDITOR.MODEL.PROPERTY";
+
+    /**
+     * The name of the file used to describe the UI. Defaults to editors.xml.
+     */
+    String getEditorFileName();
+
+    /**
+     * To signal that an editor will be provided by the connector or studio
+     */
+    boolean isProvided();
```

Comment:
it doesn't make much sense to me that this method returning true can mean many things. To me a provided editor means that Studio handles it on its own. If the model property points to a custom editor shipped with the extension then it's by definition not provided.

Maybe we should rename this to isGenerated()?